File: kotlin/kotlin-vs-java/java-example/src/main/java/org/example/ex02/Exam02.java
package org.example.ex02;

/**
 * Null 안정성
 */
public class Exam02 {

    private int a;

    public Exam02() {

        var b = 20;
        int c = 30;
        int d;

        Integer e = null;
        Integer f = 20;

        callFunction(a);
        callFunction(b);
        callFunction(c);
//        System.out.println(d);
        callFunction(e); // null point exception
        callFunction(f);
    }

    private void callFunction(Integer i) {
        var temp = i == null ? "null 값 입니다." : i;
        System.out.println(temp);
    }

    public static void main(String[] args) {
        new Exam02();
    }
}

File: kotlin/kotlin-vs-java/kotlin-example/src/main/kotlin/ex02/Exam02.kt
package ex02

/**
 * Null 안정성 및 엘비스 연산자
 */
fun main() {

    val a: Int
    val b = 10
    val c: Int = 20
    val d: Int? = null // ? => 엘비스 연산자 null 가능

//    println(a)
    callFunction(b)
    callFunction(c)
    callFunction(d)
}

fun callFunction(i: Int?) {
    // ? => null 이 올 수 있다.
    // 변수? => 변수가 null 인가?
    // 변수?. => 변수가 null 이 아닐 때
    // 변수?: => 변수가 null 일 때

    val temp = i ?: "null 값 입니다."
    println(temp)
}